presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2113,6 +2125,65 @@ private void verifySelectDistinct(QuerySpecification node, List<Expression> outp
             return orderByFields;
         }
 
+        private void analyzeLimit(Query node)
```

Comment:
These two methods are unnecessary. Aside from delegating to `analyzeLimit(node, clause)` they only check to make sure the limit is present. If you inline them, the check is unnecessary, as that's ensured by the `if` condition in the methods above.